### PR TITLE
Support user data, init script and more!

### PR DESCRIPTION
**Pull request by RoryHunter**
Support for configuring:
* Droplet user data
* Jenkins init script

Add a slave configure-entries.jelly to silence an exception when viewing the node's configuring. It's not great but it's better than nothing.

Check for an SSH connection less aggressively, and wait for the droplet to enter the ACTIVE state before doing so.

Increase default idle time to 60 minutes, since that's the minimum you'll pay for anyway.

### CODE REVIEW

File: src/main/resources/com/dubture/jenkins/digitalocean/SlaveTemplate/config.jelly [hunk 1]
```diff
@@ -47,7 +47,15 @@
         </f:entry>
 
         <f:entry title="Idle termination time" field="idleTerminationInMinutes">
-            <f:textbox default="10" />
+            <f:textbox default="60" />
```
Comment (anpieber) [inline src/main/resources/com/dubture/jenkins/digitalocean/SlaveTemplate/config.jelly]: I'm not so happy with this change. The problem is that you pay for every started hour. If the build took 10 minutes and you're waiting for 60 minutes you already pay two hours. IMHO the only correct solution would be to completely change this feature to only a radio button: 

[ ] keep alive
[ ] shutdown at next full hour

Whilte the first one would never shutdown the container, the second one will keep the container alive till the next full hour is reached and the scheduling algorithm would almost take the container with the longest remaining time to live (but those two things could be two different features). Till we've something like this in place I think the 10 minutes are the saner default choice.

File: src/main/java/com/dubture/jenkins/digitalocean/ComputerLauncher.java [hunk 2]
```diff
@@ -189,29 +189,45 @@ private int bootstrap(Connection bootstrapConn, Computer computer, PrintStream l
...
+                    case ACTIVE:
+                        break;
+
+                    case ARCHIVE:
+                    case OFF:
+                        throw new IllegalStateException("Droplet has unexpected status: " + droplet.getStatus());
+
+                    case NEW:
+                        logger.println("Droplet in NEW state, waiting for it to become ACTIVE");
+                        throw new IOException("sleep");
```
Comment (anpieber) [inline src/main/java/com/dubture/jenkins/digitalocean/ComputerLauncher.java]: tbh I dont like the approach of handling a regular Situation by an exception. sleep cont. should work fine here too, wouldn't it?

**Comment by anpieber** [comment]
Thank you so much for your work btw! I really appreciate it.

**Comment by RoryHunter** [comment]
I've reverted the timeout as you suggest - I'll see about a more clever retention strategy if I get a chance. I've also refactored the connection code quite a lot, I think it's much improved.

**Comment by anpieber** [comment]
Thank you very much again! I'll cut a release right away.

**Comment by chriswalters** [comment]
Prior to the timeout changes, my droplets were launching and remaining connected with no problem. With 0.6 I'm consistently getting a “Unexpected termination of channel” error. I tried setting the "Idle termination time" as high as 1000 (as well as empty and 0) and continued to get the error. I switched back to 0.5 and the error went away.

**Comment by RoryHunter** [comment]
I saw this too. I'll push a revert ASAP. 

-- 
Rory Hunter

**Comment by RoryHunter** [comment]
PR  #14.

**Comment by anpieber** [comment]
Released

**Pull request merged**